### ISSUE TITLE
Parse error has missing information for named invokes

### ISSUE THREAD

**Issue by rupertford** [stfc/PSyclone#94]
As reported by Chris Maynard ... In the following example, PSyclone raises an exception but does not include the name in the exception text ...

```
module alg
  use field_mod, only: field_type
  implicit none
  subroutine mini_alg
    implicit none
    type(field_type) :: f1, f2, f3
    call invoke( name = "x_times_y", &
                 X_times_Y(f1, f2, f3)  &
                  )
    call invoke( name = "x_times_y", &
                 X_times_Y(f2, f3, f1)  &
                  )
  end subroutine mini_alg
end module alg
```

When he runs PSyclone over this it says …

 `Parse Error: Found multiple named invoke()'s with the same name: ""`

He would like the actual name, the line number and the file name.

**Comment by arporter** [comment]
Created branch `name_case_fix` for this work and to tackle #96.

**Comment by rupertford** [comment]
Fixed by Andy in PR #97. Closing this issue
